? checkLineLengthIgnore.patch
? metadata/com/puppycrawl/tools/checkstyle/checks/sizes/checkstyle-all-5.1.jar
Index: metadata/com/puppycrawl/tools/checkstyle/checks/sizes/checkstyle-metadata.properties
===================================================================
RCS file: /cvsroot/eclipse-cs/net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/sizes/checkstyle-metadata.properties,v
retrieving revision 1.2
diff -r1.2 checkstyle-metadata.properties
1,36c1,40
< 
< AnonInnerLength.desc = Checks for long anonymous inner classes.<br/>\r\nRationale: If an anonymous inner class becomes very long it is hard to understand and to see the flow of the method where the class is defined. Therefore long anonymous inner classes should usually be refactored into a named inner class. See also Bloch, Effective Java, p. 93. 
< AnonInnerLength.max  = maximum allowable number of lines.
< AnonInnerLength.name = Anonymous inner classes lengths
< 
< ExecutableStatementCount.desc   = Restricts the number of executable statements to a specified limit
< ExecutableStatementCount.max    = Maximum allowed parameters.
< ExecutableStatementCount.name   = Executable Statement Size
< ExecutableStatementCount.tokens = Tokens to check.
< 
< FileLength.desc = Checks for long source files.<br/>\r\nRationale: If a source file becomes very long it is hard to understand. Therefore long classes should usually be refactored into several individual classes that focus on a specific task. 
< FileLength.max  = Maximum number of lines allowed.
< FileLength.name = Maximum File Length
< 
< LineLength.desc          = Checks for long lines.<br/>\r\nRationale: Long lines are hard to read in printouts or if developers have limited screen space for the source code, e.g. if the IDE displays additional information like project tree, class hierarchy, etc. 
< LineLength.ignorePattern = Pattern for lines to ignore.
< LineLength.max           = Maximum allowed line length.
< LineLength.name          = Maximum Line Length
< LineLength.tabWidth      = Number of characters to count for each tab character.
< 
< MethodLength.countEmpty = whether to count empty lines and single line comments of the form //
< MethodLength.desc       = Checks for long methods and constructors.<br/>\r\nRationale: If a method becomes very long it is hard to understand. Therefore long methods should usually be refactored into several individual methods that focus on a specific task. 
< MethodLength.max        = Maximum number of lines allowed.
< MethodLength.name       = Maximum Method Length
< MethodLength.tokens     = Tokens to check.
< 
< OuterTypeNumber.desc = Checks for the number of types declared at the outer (or root) level in a file.<br/>\r\n<br/>\r\nRationale: It is considered good practice to only define one outer type per file. 
< OuterTypeNumber.max  = maximum allowable number of outer types
< OuterTypeNumber.name = Outer Type Number
< 
< ParameterNumber.desc   = Checks the number of parameters of a method or constructor.
< ParameterNumber.max    = Maximum allowed parameters.
< ParameterNumber.name   = Maximum Parameters
< ParameterNumber.tokens = Tokens to check.
< 
< Sizes.group = Size Violations
---
> 
> AnonInnerLength.desc = Checks for long anonymous inner classes.<br/>\r\nRationale: If an anonymous inner class becomes very long it is hard to understand and to see the flow of the method where the class is defined. Therefore long anonymous inner classes should usually be refactored into a named inner class. See also Bloch, Effective Java, p. 93. 
> AnonInnerLength.max  = maximum allowable number of lines.
> AnonInnerLength.name = Anonymous inner classes lengths
> 
> ExecutableStatementCount.desc   = Restricts the number of executable statements to a specified limit
> ExecutableStatementCount.max    = Maximum allowed parameters.
> ExecutableStatementCount.name   = Executable Statement Size
> ExecutableStatementCount.tokens = Tokens to check.
> 
> FileLength.desc = Checks for long source files.<br/>\r\nRationale: If a source file becomes very long it is hard to understand. Therefore long classes should usually be refactored into several individual classes that focus on a specific task. 
> FileLength.max  = Maximum number of lines allowed.
> FileLength.name = Maximum File Length
> 
> LineLength.desc          = Checks for long lines.<br/>\r\nRationale: Long lines are hard to read in printouts or if developers have limited screen space for the source code, e.g. if the IDE displays additional information like project tree, class hierarchy, etc. 
> LineLength.ignorePattern = Pattern for lines to ignore.
> LineLength.max           = Maximum allowed line length.
> LineLength.name          = Maximum Line Length
> LineLength.tabWidth      = Number of characters to count for each tab character.
> LineLength.IgnoreClass = Check class declaration length.
> LineLength.IgnoreConstructor = Check constructor declaration length.
> LineLength.IgnoreField = Check field declaration length.
> LineLength.IgnoreMethod = Check method declaration length.
> 
> MethodLength.countEmpty = whether to count empty lines and single line comments of the form //
> MethodLength.desc       = Checks for long methods and constructors.<br/>\r\nRationale: If a method becomes very long it is hard to understand. Therefore long methods should usually be refactored into several individual methods that focus on a specific task. 
> MethodLength.max        = Maximum number of lines allowed.
> MethodLength.name       = Maximum Method Length
> MethodLength.tokens     = Tokens to check.
> 
> OuterTypeNumber.desc = Checks for the number of types declared at the outer (or root) level in a file.<br/>\r\n<br/>\r\nRationale: It is considered good practice to only define one outer type per file. 
> OuterTypeNumber.max  = maximum allowable number of outer types
> OuterTypeNumber.name = Outer Type Number
> 
> ParameterNumber.desc   = Checks the number of parameters of a method or constructor.
> ParameterNumber.max    = Maximum allowed parameters.
> ParameterNumber.name   = Maximum Parameters
> ParameterNumber.tokens = Tokens to check.
> 
> Sizes.group = Size Violations
Index: metadata/com/puppycrawl/tools/checkstyle/checks/sizes/checkstyle-metadata.xml
===================================================================
RCS file: /cvsroot/eclipse-cs/net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/sizes/checkstyle-metadata.xml,v
retrieving revision 1.5
diff -r1.5 checkstyle-metadata.xml
1,109c1,126
< <?xml version="1.0" encoding="UTF-8"?>
< <!DOCTYPE checkstyle-metadata PUBLIC
< "-//eclipse-cs//DTD Check Metadata 1.1//EN"
< "http://eclipse-cs.sourceforge.net/dtds/checkstyle-metadata_1_1.dtd">
< <checkstyle-metadata>
<     <rule-group-metadata name="%Sizes.group" priority="500">
< 
<         <rule-metadata name="%AnonInnerLength.name" internal-name="AnonInnerLength" parent="TreeWalker">
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.AnonInnerLengthCheck"/>
<             <description>%AnonInnerLength.desc</description>
<             <property-metadata name="max" datatype="Integer" default-value="20">
<                 <description>%AnonInnerLength.max</description>
<             </property-metadata>
<             <message-key key="maxLen.anonInner"/>
<         </rule-metadata>
< 
<         <rule-metadata name="%ExecutableStatementCount.name" internal-name="ExecutableStatementCount"
<             parent="TreeWalker">
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.ExecutableStatementCountCheck"/>
<             <description>%ExecutableStatementCount.desc</description>
<             <property-metadata name="max" datatype="Integer" default-value="30">
<                 <description>%ExecutableStatementCount.max</description>
<             </property-metadata>
<             <property-metadata name="tokens" datatype="MultiCheck"
<                 default-value="CTOR_DEF,METHOD_DEF,INSTANCE_INIT,STATIC_INIT">
<                 <description>%ExecutableStatementCount.tokens</description>
<                 <enumeration>
<                     <property-value-option value="INSTANCE_INIT"/>
<                     <property-value-option value="STATIC_INIT"/>
<                     <property-value-option value="METHOD_DEF"/>
<                     <property-value-option value="CTOR_DEF"/>
<                 </enumeration>
<             </property-metadata>
<             <message-key key="executableStatementCount"/>
<         </rule-metadata>
< 
<         <rule-metadata name="%FileLength.name" internal-name="FileLength" parent="Checker">
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.FileLengthCheck"/>
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.FileLengthCheck"/>
<             <description>%FileLength.desc</description>
<             <property-metadata name="max" datatype="Integer" default-value="2000">
<                 <description>%FileLength.max</description>
<             </property-metadata>
<             <message-key key="maxLen.file"/>
<         </rule-metadata>
< 
<         <rule-metadata name="%LineLength.name" internal-name="LineLength" parent="TreeWalker">
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck"/>
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.LineLengthCheck"/>
<             <description>%LineLength.desc</description>
<             <property-metadata name="ignorePattern" datatype="Regex" default-value="^$">
<                 <description>%LineLength.ignorePattern</description>
<             </property-metadata>
<             <property-metadata name="max" datatype="Integer" default-value="80">
<                 <description>%LineLength.max</description>
<             </property-metadata>
<             <property-metadata name="tabWidth" datatype="Integer" default-value="8">
<                 <description>%LineLength.tabWidth</description>
<             </property-metadata>
<             <message-key key="maxLineLen"/>
<         </rule-metadata>
< 
<         <rule-metadata name="%MethodLength.name" internal-name="MethodLength" parent="TreeWalker">
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.MethodLengthCheck"/>
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.MethodLengthCheck"/>
<             <description>%MethodLength.desc</description>
<             <property-metadata name="max" datatype="Integer" default-value="150">
<                 <description>%MethodLength.max</description>
<             </property-metadata>
<             <property-metadata name="countEmpty" datatype="Boolean" default-value="true">
<                 <description>%MethodLength.countEmpty</description>
<             </property-metadata>
<             <property-metadata name="tokens" datatype="MultiCheck" default-value="METHOD_DEF,CTOR_DEF">
<                 <description>%MethodLength.tokens</description>
<                 <enumeration>
<                     <property-value-option value="METHOD_DEF"/>
<                     <property-value-option value="CTOR_DEF"/>
<                 </enumeration>
<             </property-metadata>
<             <message-key key="maxLen.method"/>
<         </rule-metadata>
< 
<         <rule-metadata name="%ParameterNumber.name" internal-name="ParameterNumber" parent="TreeWalker">
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.ParameterNumberCheck"/>
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.ParameterNumberCheck"/>
<             <description>%ParameterNumber.desc</description>
<             <property-metadata name="max" datatype="Integer" default-value="7">
<                 <description>%ParameterNumber.max</description>
<             </property-metadata>
<             <property-metadata name="tokens" datatype="MultiCheck" default-value="METHOD_DEF,CTOR_DEF">
<                 <description>%ParameterNumber.tokens</description>
<                 <enumeration>
<                     <property-value-option value="METHOD_DEF"/>
<                     <property-value-option value="CTOR_DEF"/>
<                 </enumeration>
<             </property-metadata>
<             <message-key key="maxParam"/>
<         </rule-metadata>
< 
<         <rule-metadata name="%OuterTypeNumber.name" internal-name="OuterTypeNumber" parent="TreeWalker">
<             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.OuterTypeNumber"/>
<             <description>%OuterTypeNumber.desc</description>
<             <property-metadata name="max" datatype="Integer" default-value="1">
<                 <description>%OuterTypeNumber.max</description>
<             </property-metadata>
<             <message-key key="maxOuterTypes"/>
<         </rule-metadata>
<     </rule-group-metadata>
< </checkstyle-metadata>
---
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE checkstyle-metadata PUBLIC
> "-//eclipse-cs//DTD Check Metadata 1.1//EN"
> "http://eclipse-cs.sourceforge.net/dtds/checkstyle-metadata_1_1.dtd">
> <checkstyle-metadata>
>     <rule-group-metadata name="%Sizes.group" priority="500">
> 
>         <rule-metadata name="%AnonInnerLength.name" internal-name="AnonInnerLength" parent="TreeWalker">
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.AnonInnerLengthCheck"/>
>             <description>%AnonInnerLength.desc</description>
>             <property-metadata name="max" datatype="Integer" default-value="20">
>                 <description>%AnonInnerLength.max</description>
>             </property-metadata>
>             <message-key key="maxLen.anonInner"/>
>         </rule-metadata>
> 
>         <rule-metadata name="%ExecutableStatementCount.name" internal-name="ExecutableStatementCount"
>             parent="TreeWalker">
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.ExecutableStatementCountCheck"/>
>             <description>%ExecutableStatementCount.desc</description>
>             <property-metadata name="max" datatype="Integer" default-value="30">
>                 <description>%ExecutableStatementCount.max</description>
>             </property-metadata>
>             <property-metadata name="tokens" datatype="MultiCheck"
>                 default-value="CTOR_DEF,METHOD_DEF,INSTANCE_INIT,STATIC_INIT">
>                 <description>%ExecutableStatementCount.tokens</description>
>                 <enumeration>
>                     <property-value-option value="INSTANCE_INIT"/>
>                     <property-value-option value="STATIC_INIT"/>
>                     <property-value-option value="METHOD_DEF"/>
>                     <property-value-option value="CTOR_DEF"/>
>                 </enumeration>
>             </property-metadata>
>             <message-key key="executableStatementCount"/>
>         </rule-metadata>
> 
>         <rule-metadata name="%FileLength.name" internal-name="FileLength" parent="Checker">
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.FileLengthCheck"/>
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.FileLengthCheck"/>
>             <description>%FileLength.desc</description>
>             <property-metadata name="max" datatype="Integer" default-value="2000">
>                 <description>%FileLength.max</description>
>             </property-metadata>
>             <message-key key="maxLen.file"/>
>         </rule-metadata>
> 
>         <rule-metadata name="%LineLength.name" internal-name="LineLength" parent="TreeWalker">
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck"/>
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.LineLengthCheck"/>
>             <description>%LineLength.desc</description>
>             <property-metadata name="ignorePattern" datatype="Regex" default-value="^$">
>                 <description>%LineLength.ignorePattern</description>
>             </property-metadata>
>             <property-metadata name="max" datatype="Integer" default-value="80">
>                 <description>%LineLength.max</description>
>             </property-metadata>
>             <property-metadata name="tabWidth" datatype="Integer" default-value="8">
>                 <description>%LineLength.tabWidth</description>
> </property-metadata>
> 
> 		<property-metadata name="ignoreClass" datatype="Boolean" default-value="false">
>                 <description>%LineLength.IgnoreClass</description>
> </property-metadata>
> 
> 	    <property-metadata name="ignoreConstructor" datatype="Boolean" default-value="false">
>                 <description>%LineLength.IgnoreConstructor</description>
> </property-metadata>
> 
> 	    <property-metadata name="ignoreField" datatype="Boolean" default-value="false">
>                 <description>%LineLength.IgnoreField</description>
> </property-metadata>
> 
> 	    <property-metadata name="ignoreMethod" datatype="Boolean" default-value="false">
>                 <description>%LineLength.IgnoreMethod</description>
> </property-metadata>
> 
> 	    <message-key key="maxLineLen"/>
>         </rule-metadata>
> 
>         <rule-metadata name="%MethodLength.name" internal-name="MethodLength" parent="TreeWalker">
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.MethodLengthCheck"/>
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.MethodLengthCheck"/>
>             <description>%MethodLength.desc</description>
>             <property-metadata name="max" datatype="Integer" default-value="150">
>                 <description>%MethodLength.max</description>
>             </property-metadata>
>             <property-metadata name="countEmpty" datatype="Boolean" default-value="true">
>                 <description>%MethodLength.countEmpty</description>
>             </property-metadata>
>             <property-metadata name="tokens" datatype="MultiCheck" default-value="METHOD_DEF,CTOR_DEF">
>                 <description>%MethodLength.tokens</description>
>                 <enumeration>
>                     <property-value-option value="METHOD_DEF"/>
>                     <property-value-option value="CTOR_DEF"/>
>                 </enumeration>
>             </property-metadata>
>             <message-key key="maxLen.method"/>
>         </rule-metadata>
> 
>         <rule-metadata name="%ParameterNumber.name" internal-name="ParameterNumber" parent="TreeWalker">
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.ParameterNumberCheck"/>
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.ParameterNumberCheck"/>
>             <description>%ParameterNumber.desc</description>
>             <property-metadata name="max" datatype="Integer" default-value="7">
>                 <description>%ParameterNumber.max</description>
>             </property-metadata>
>             <property-metadata name="tokens" datatype="MultiCheck" default-value="METHOD_DEF,CTOR_DEF">
>                 <description>%ParameterNumber.tokens</description>
>                 <enumeration>
>                     <property-value-option value="METHOD_DEF"/>
>                     <property-value-option value="CTOR_DEF"/>
>                 </enumeration>
>             </property-metadata>
>             <message-key key="maxParam"/>
>         </rule-metadata>
> 
>         <rule-metadata name="%OuterTypeNumber.name" internal-name="OuterTypeNumber" parent="TreeWalker">
>             <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.sizes.OuterTypeNumber"/>
>             <description>%OuterTypeNumber.desc</description>
>             <property-metadata name="max" datatype="Integer" default-value="1">
>                 <description>%OuterTypeNumber.max</description>
>             </property-metadata>
>             <message-key key="maxOuterTypes"/>
>         </rule-metadata>
>     </rule-group-metadata>
> </checkstyle-metadata>
